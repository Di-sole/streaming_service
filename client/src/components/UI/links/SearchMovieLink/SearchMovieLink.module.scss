.container {
    display: flex;
    color: #a5a1b2;
    width: 100%;

    &__icon {
        width: 22px;
        height: 22px;
        margin: 0 12px 0 0;
    }

    &__name {
        font-weight: 700;
        font-size: 15px;
        line-height: 20px;
        margin: 0 0 5px 0;
        transition: 0.2s;

        & > span {
            color: #fff;
        }
    }

    &:hover {
        .container__name {
            color: #ff0000;
        }
    }
}
