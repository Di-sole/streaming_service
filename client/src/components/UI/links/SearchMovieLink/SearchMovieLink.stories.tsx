import { Meta, StoryObj } from '@storybook/react';
import Link from './SearchMovieLink';
import './SearchMovieLink.module.scss';

const meta: Meta<typeof Link> = {
    title: 'Links/SearchMovieLink',
    component: Link,
    decorators: [
        (Story) => (
            <div style={{ background: '#100e19' }}>
                <Story />
            </div>
        ),
    ],
};

export default meta;

type Story = StoryObj<typeof Link>;

export const SearchMovieLink: Story = {
    args: {
        href: '/',
        name: 'Фильм',
        year: 1900,
        value: '',
    },
};
