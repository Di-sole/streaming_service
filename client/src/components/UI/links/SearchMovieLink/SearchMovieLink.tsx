import Image from 'next/image';
import styles from './SearchMovieLink.module.scss';
import Link from 'next/link';
import movieIcon from '../../../../../public/icons/movie/movie.svg';

interface SearchMovieItemProps {
    href: string;
    name: string;
    year: number;
    value: string;
    onClick?: (e: any) => void;
}

const SearchMovieItem = ({ href, name, year, value, onClick }: SearchMovieItemProps) => {
    let starOfLine = name.toLowerCase().indexOf(value.toLowerCase());
    let start = name.slice(0, starOfLine);
    let search = name.slice(starOfLine, starOfLine + value.length);
    let end = name.slice(starOfLine + value.length);

    return (
        <Link className={styles.container} href={href} onClick={onClick}>
            <Image className={styles.container__icon} src={movieIcon} alt="" />

            <div className={styles.container__description}>
                <h2 className={styles.container__name}>
                    {start}
                    <span>{search}</span>
                    {end}
                </h2>
                <p className={styles.container__year}>{year}</p>
            </div>
        </Link>
    );
};

export default SearchMovieItem;
