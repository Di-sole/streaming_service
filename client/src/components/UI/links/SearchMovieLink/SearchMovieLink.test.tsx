import { render, screen } from '@testing-library/react';
import SearchMovieLink from './SearchMovieLink';

describe('SEARCHMOVIELINK SNAPSHOTS TESTS', () => {
    test('SearchMovieLink should not be changed', () => {
        render(<SearchMovieLink href="/" name="text" year={1900} value="" />);
        const link = screen.getByRole('link');
        expect(link).toMatchSnapshot();
    });
});
