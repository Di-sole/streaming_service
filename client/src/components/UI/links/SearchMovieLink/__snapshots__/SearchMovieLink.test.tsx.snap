// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SEARCHMOVIELINK SNAPSHOTS TESTS SearchMovieLink should not be changed 1`] = `
<a
  class="container"
  href="/"
>
  <img
    alt=""
    class="container__icon"
    data-nimg="1"
    decoding="async"
    height="40"
    loading="lazy"
    src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
    srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
    style="color: transparent;"
    width="40"
  />
  <div
    class="container__description"
  >
    <h2
      class="container__name"
    >
      <span />
      text
    </h2>
    <p
      class="container__year"
    >
      1900
    </p>
  </div>
</a>
`;
